lib/rules/function-url-quotes/index.mjs

```diff
@@ -54,48 +58,72 @@ const rule = (primary, secondaryOptions, context) => {
 		root.walkDecls(checkDeclParams);
 
 		/**
-		 * @param {import('postcss').Declaration} decl
+		 * @param {Declaration|AtRule} node
+		 * @param {FunctionNode} fn
+		 * @param {keyof messages} messageType
+		 * @param {ParsedValue} parsedValue
+		 */
+		function getFix(node, fn, messageType, parsedValue) {
+			return () => {
+				switch (messageType) {
+					case 'expected':
+						addQuotes(fn);
+						break;
+					case 'rejected':
+						removeQuotes(fn);
+						break;
+				}
+
+				switch (node.type) {
+					case 'decl':
+						node.value = parsedValue.toString();
+						break;
+					case 'atrule':
+						node.params = parsedValue.toString();
+						break;
+				}
+			};
+		}
+
+		/**
+		 * @param {AtRule|Declaration} node
+		 * @param {string} source
+		 * @param {number} startIndex
 		 */
+		function complain(node, source, startIndex) {
+			functionArgumentsSearch(source, /^url$/i, (args, index, funcNode, parsedValue) => {
+				const object = checkArgs(args, startIndex + index, funcNode);
+
+				if (object) {
+					const { messageType, ...problem } = object;
+					const fix = getFix(node, funcNode, messageType, parsedValue);
+
+					report({ ...problem, node, fix });
```

Comment:
[suggest] I also want to simplify `checkArgs()` and make `report()` arguments clearer for readability. The `checkArgs()` logic actually does not relate to `funcNode`, `ruleName`, nor `result`. For example:

```js
const object = checkArgs(args, startIndex + index);

if (object) {
	const { messageType, index, endIndex } = object;
	const message = messages[messageType](funcNode.value.toLowerCase());
	const fix = getFix(node, funcNode, messageType, parsedValue);
	
	report({ ruleName, result, node, index, endIndex, message, fix });
}
```

Reply:
`index` clashes with other `index`; Ill use `...rest`.